indra/newview/gltf/llgltfloader.cpp

```diff
@@ -290,12 +310,9 @@ void LLGLTFLoader::addModelToScene(
                 materials[model->mMaterialList[i]] = LLImportMaterial();
             }
         }
-        std::string base_name = model_name;
-        if (model->mSubmodelID > 0)
-        {
-            base_name += (char)((int)'a' + model->mSubmodelID);
-        }
-        mScene[transformation].push_back(LLModelInstance(model, base_name, transformation, materials));
+        // Compose instance name from model label and add lod suffix
+        std::string instance_name = model->mLabel + lod_suffix[mLod];
```

Comment:
suffix should not get into mScene. It should only be present in model->mLabel for model matching to work.
std::string instance_name = model->mLabel;
model->mLabel += lod_suffix[mLod];
mScene[transformation].push_back(LLModelInstance(model, instance_name, transformation, materials));